### DISCUSSION TITLE
New Chart from different worksheet?

### DISCUSSION THREAD

**Discussion by RipFence** [dfinke/ImportExcel#1577] [Q&A] [answered]
Is it possible to create a chart from data on a different worksheet than the data?  For example, my data is on the "Data" worksheet, but I want to have the chart on the "Summary" worksheet.  I've tried using PivotTables, Ranges, etc and no luck.  I'm only successful when the Chart is on the same page as its referenced data.

**Comment by dfinke** [comment, marked as answer]
@RipFence Good question. If you're up for it, this `as is` would make a great PR from you as an example. 😉

```powershell
$data = ConvertFrom-Csv @"
Region,State,Units,Price
West,Texas,927,923.71
North,Tennessee,466,770.67
East,Florida,520,458.68
East,Maine,828,661.24
West,Virginia,465,053.58
North,Missouri,436,235.67
South,Kansas,214,992.47
North,North Dakota,789,640.72
South,Delaware,712,508.55
"@

$xlfile = "$PSScriptRoot\spike.xlsx"
Remove-Item $xlfile -ErrorAction SilentlyContinue

$xlpkg = $data | Export-Excel $xlfile -WorksheetName Data -AutoNameRange -PassThru

$null = Add-Worksheet -ExcelPackage $xlpkg -WorksheetName Summary -Activate

$params = @{
    Worksheet = $xlpkg.Summary
    Title     = "Sales by Region"
    ChartType = 'ColumnClustered'
    
    # XRange    = "Data!A2:A10"
    # YRange    = "Data!C2:C10"

    XRange    = 'Data!Region'
    YRange    = 'Data!Units'
}

Add-ExcelChart  @params

Close-ExcelPackage $xlpkg -Show
```

**Reply by RipFence** [reply]
LOL, you do all the work and I submit the PR?  Deal 😄.  Thanks for all of the hard work on this module, Doug.  It's a very complete and useful tool for automated reporting.

**Reply by dfinke** [reply]
@RipFence you found the "issue". Had not covered that. It took me a bit to get my head around it. 

Your question made the module better with examples.

So yes, PR is in order :)
